lib/commands/file.js

```diff
@@ -95,24 +97,39 @@ options(program, 'files-upload <file> <datasetID>')
       const fileSize = stats['size'];
       const fileName = file.startsWith('./') ? file.slice(2) : file;
 
-      if (fileSize > MULTIPART_MIN_SIZE) {
-        const response = await post(options, UPLOADS_URL, {
-          name: fileName,
-          datasetId: dataset,
-          overwrite: options.overwrite
-        });
+      let uploadResponse = null;
+
+      try {
+        if (fileSize > MULTIPART_MIN_SIZE) {
+          uploadResponse = await post(options, UPLOADS_URL, {
+            name: fileName,
+            datasetId: dataset,
+            overwrite: options.overwrite
+          });
+
+          await multipartUpload(options, uploadResponse.data.uploadId, file, fileSize);
+          console.log(chalk.green(`Upload complete: ${fileName}, ID: ${uploadResponse.data.id}`));
+        } else {
+          uploadResponse = await post(options, FILES_URL, {
+            name: fileName,
+            datasetId: dataset,
+            overwrite: options.overwrite
+          });
+
+          await upload(uploadResponse.data.uploadUrl, file, fileSize);
+          console.log(chalk.green(`Upload complete: ${fileName}, ID: ${uploadResponse.data.id}`));
+        }
+      } catch (error) {
+        if (_get(error, 'response.data.error', '').indexOf('already exists') > -1) {
+          console.log(chalk.yellow(`Ignoring already uploaded file: ${fileName}`));
+        } else {
+          throw error;
+        }
+      }
 
-        await multipartUpload(options, response.data.uploadId, file, fileSize);
-        console.log(chalk.green(`Upload complete: ${fileName}, ID: ${response.data.id}`));
-      } else {
-        const response = await post(options, FILES_URL, {
-          name: fileName,
-          datasetId: dataset,
-          overwrite: options.overwrite
-        });
-
-        await upload(response.data.uploadUrl, file, fileSize);
-        console.log(chalk.green(`Upload complete: ${fileName}, ID: ${response.data.id}`));
+      if (options.deleteAfterUpload) {
+        // eslint-disable-next-line security/detect-non-literal-fs-filename
```

Comment:
There needs to be a double check here to verify the file really exists server side before deleting. At the very least it should check that the file size matches.... it'd even better if we had a checksum to verify.

Reply:
I thought about a checksum as well, but am thinking let's keep the PRs simple/small.  File size is a good idea.  Due to eventual consistency, we'll need to introduce a polling solution to verify each file.  I was going to do this as a separate PR - but can work it into this one.